src/lib/x509/x509_ext.h

```diff
@@ -500,6 +500,66 @@ class OCSP_NoCheck final : public Certificate_Extension {
       void decode_inner(const std::vector<uint8_t>&) override;
 };
 
+/**
+* TNAuthList extension
+*
+* RFC8226 Secure Telephone Identity Credentials
+*   https://www.rfc-editor.org/rfc/rfc8226#section-9
+*/
+class BOTAN_PUBLIC_API(3, 6) TNAuthList final : public Certificate_Extension {
+   public:
+      class TNEntry final : public ASN1_Object {
+         public:
+            enum Type { TN_ServiceProviderCode = 0, TN_TelephoneNumberRange, TN_TelephoneNumber };
+
+            struct TelephoneNumberRange {
+                  ASN1_String start;  //TelephoneNumber (IA5String)
+                  size_t count;       //2..MAX
+            };
+
+            TNEntry() = default;
+            virtual ~TNEntry() = default;
+
+            void encode_into(DER_Encoder&) const override;
+            void decode_from(class BER_Decoder& from) override;
+
+            Type get_type() const { return type; }
+
+            const std::string& getServiceProviderCode() const { return spc.value(); }
+
+            const std::vector<TelephoneNumberRange>& getTelephoneNumberRange() const { return range; }
+
+            const std::string& getTelephoneNumber() const { return one.value(); }
```

Comment:
Typically, getters in Botan aren't prefixed and methods are in snake case:

```suggestion
            Type type() const { return type; }

            const std::string& service_provider_code() const { return spc.value(); }

            const std::vector<TelephoneNumberRange>& telephone_number_range() const { return range; }

            const std::string& telephone_number() const { return one.value(); }
```